site/public/js/user-profile.js

```diff
@@ -121,6 +128,50 @@ function updateUserProfilePhoto () {
   return false;
 }
 
+function updateUserSecondaryEmail () {
+    const second_email = $('#user-secondary-email-change');
+    const second_email_notify = $('#user-secondary-email-notify-change');
+    if (second_email.data('current-second-email') === second_email.val() && second_email_notify.get(0).checked === (second_email_notify.data('current-second-email-notify') === 1)) {
+        displayErrorMessage('No changes detected to secondary email');
+    }
+    else {
+        let data = new FormData();
+        data.append('csrf_token', csrfToken);
+        data.append('secondary_email', second_email.val());
+        data.append('secondary_email_notify', second_email_notify.get(0).checked);
+        let url = buildUrl(['user_profile', 'change_secondary_email']);
+        $.ajax({
+            url,
+            type: "POST",
+            data,
+            processData: false,
+            contentType: false,
+            success: function(res) {
+                const response = JSON.parse(res);
+                if (response.status === "success") {
+                    const { data } = response;
+                    displaySuccessMessage(data.message);
+                    $('#secondary-email-row .value').text(data.secondary_email);
+                    $('#secondary-email-notify-row .value').text(data.secondary_email_notify);
+                    second_email.data('current-second-email', data.secondary_email);
+                    second_email_notify.data('current-second-email-notify', data.secondary_email_notify === "True" ? 1 : 0);
+                }
+                else if (response.status === "error") {
+                    displayErrorMessage("Some went wrong while updating secondary email address!");
```

Comment:
Typo:
```suggestion
                    displayErrorMessage("Something went wrong while updating secondary email address!");
```

Though, could we give a more meaningful error message in this case, like why updating the secondary email address failed (like bad validation, etc.)?

Reply:
I have changed this to just simply display the error message the backend responds with.